build/storage/tests/it/ptf_tests/system_tools/test_platform.py

```diff
@@ -96,65 +101,99 @@ def _install_docker(self):
         if not self._set_security_policies():
-            raise Exception('Security polices is not setting properly')
+            raise DependencesException("Security polices is not setting properly")
         if not self._is_docker():
-            raise Exception('Docker is not setting properly')
+            raise DependencesException("Docker is not setting properly")
 
-    # TODO: after testing restore settings
     def set_system_setup(self):
         """Overwrite this method in specific platform if you don't want set all setup"""
         self._change_vmlinuz()
         if not self._is_docker():
             self._install_docker()
 
+    def get_pid_from_port(self, port: int):
+        return self.terminal.execute(
+            f"sudo netstat -anop | grep -Po ':{port}\s.*LISTEN.*?\K\d+(?=/)' || true"
+        )
 
-class StorageTargetPlatform(BaseTestPlatform):
+    def kill_process_from_port(self, port: int):
+        """Raise error if there is no process in specific port"""
+        pid = self.get_pid_from_port(port)
+        self.terminal.execute(f"sudo kill -9 {pid}")
 
-    def __init__(self):
-        config = StorageTargetConfig()
-        terminal = SSHTerminal(config)
-        super().__init__(terminal)
+    def clean(self):
+        self.cmd_sender.stop()
+        self.docker.delete_containers()
+
+    def is_port_free(self, port):
+        return not bool(
+            self.terminal.execute(f"sudo netstat -anop | grep ':{port} ' || true")
+        )
+
+    def is_app_listening_on_port(self, app_name, port):
+        out = self.terminal.execute(f"sudo netstat -anop | grep ':{port} ' || true")
+        return "spdk_tgt" in out
+
+
+class StorageTargetPlatform(BaseTestPlatform):
+    def __init__(self, run=True):
+        super().__init__(SSHTerminal(StorageTargetConfig()), run)
+        if run:
+            self.docker.add_container(
+                "storage_target_container", StorageTargetContainer()
+            )
 
-    # TODO add implementation
     def create_subsystem(self):
-        pass
+        return self.cmd_sender.create_subsystem()
 
-    # TODO add implementation
-    def create_ramdrive(self):
-        return 'Guid'
+    def create_ramdrives(self, number):
+        return self.cmd_sender.create_ramdrives(number)
 
 
 class IPUStoragePlatform(BaseTestPlatform):
+    def __init__(self, run=True):
+        super().__init__(SSHTerminal(IPUStorageConfig()), run)
+        if run:
+            self.docker.add_container("ipu_storage_container", IPUStorageContainer())
+
+    def create_virtio_blk_devices(self, volumes_ids, physical_ids=None):
+        physical_ids = (
+            physical_ids if physical_ids is not None else range(len(volumes_ids))
+        )
+        device_handles = []
+        for volume_id, physical_id in zip(volumes_ids, physical_ids):
+            device_handles.append(
+                self.cmd_sender.create_virtio_blk_device(volume_id, physical_id)
+            )
+        return device_handles
```

Comment:
For ipu platform `clean`, we need to be careful to delete all allocated devices. Can we destroy all allocated virtio-blk devices here or at least place TODO ?
It is ok for KVM just to kill containers, but unfortunately, it is not applicable for HW platforms

Reply:
TODO